genesis/process/metaGenesisBlockCreator.go

```diff
@@ -500,15 +499,29 @@ func deploySystemSmartContracts(
 		return bytes.Compare(systemSCAddresses[i], systemSCAddresses[j]) < 0
 	})
 
+	deploySystemSCTxs := make([]data.TransactionHandler, 0)
+
 	for _, address := range systemSCAddresses {
-		tx.SndAddr = address
+		tx := &transaction.Transaction{
+			Nonce:     0,
+			Value:     big.NewInt(0),
+			RcvAddr:   rcvAddress,
+			SndAddr:   address,
+			GasPrice:  0,
+			GasLimit:  math.MaxUint64,
+			Data:      []byte(deployTxData),
+			Signature: nil,
```

Comment:
shouldn't this have the genesis signature constant?

Reply:
The genesis signature constant was used (at creation) only for minting transactions, which I create manually. 
I did not update the already "created" transactions, it will affect the hash; for staking, delegation, deploySC txs the `Signature` will be updated later on, in accountsParser